step/v2/step.go

```diff
@@ -243,3 +253,23 @@ func (p *pipeline) errExit(err error) {
 	p.failed = err
 	runtime.Goexit()
 }
+
+// cast performs a type cast from src to T.
+//
+// Unlike src.(T), this cast is valid when casting from an untyped nil to a typed nil.
+func cast[T any](src any) T {
+	if src != nil {
+		return src.(T)
+	}
+	var t T
+	typ := reflect.TypeOf(t)
```

Comment:
I'm curious as to the specific use case - I assume a very particular nil check after the function returns?

Reply:
I simplified that function. Basically, `maybeNil.(T)` panics when `maybeNil` is in fact `nil`.